index.js

```diff
@@ -55,25 +78,51 @@ try {
 
     const installPath = installPathList[installPathList.length - 1]
     console.log(`install: ${installPath}`)
+    return installPath
+}
+
+function getVSDevCmdArgs(inputs) {
+    // Default to the native processor as the host architecture
+    // vsdevcmd accepts both amd64 and x64
+    const hostArch = inputs.host_arch || process.env['PROCESSOR_ARCHITECTURE'].toLowerCase() // amd64, x86 or arm64
+
+    // Default to the host architecture as the target architecture
+    const arch = inputs.arch || hostArch
+
+    const args = [
+        `-host_arch=${hostArch}`,
+        `-arch=${arch}`
+    ]
+
+    if (inputs.toolsetVersion)
+        vsDevCmdArgs.push(`-vcvars_ver=${toolsetVersion}`)
+    if (inputs.winsdk)
+        args.push(`-winsdk=${winsdk}`)
+
+    return args
+}
+
+try {
+    // this job has nothing to do on non-Windows platforms
+    if (process.platform != 'win32') {
+        process.exit(0)
+    }
+
+    var inputs = getInputs()
+
+    const installPath = findVSInstallDir(inputs)
     core.setOutput('install_path', installPath)
 
     const vsDevCmdPath = path.win32.join(installPath, 'Common7', 'Tools', 'vsdevcmd.bat')
     console.log(`vsdevcmd: ${vsDevCmdPath}`)
 
-    const vsDevCmdArgs = [ vsDevCmdPath, `-arch=${arch}` ]
-    if (hostArch != '')
-        vsDevCmdArgs.push(`-host_arch=${hostArch}`)
-    if (toolsetVersion != '')
-        vsDevCmdArgs.push(`-vcvars_ver=${toolsetVersion}`)
-    if (winsdk != '')
-        vsDevCmdArgs.push(`-winsdk=${winsdk}`)
-    
-    const cmdArgs = [ '/q', '/k'].concat(vsDevCmdArgs, ['&&', 'set'])
-
+    const vsDevCmdArgs = getVSDevCmdArgs(inputs)
+    const cmdArgs = [].concat(['/q', '/k', vsDevCmdPath], vsDevCmdArgs, ['&&', 'set'])
```

Comment:
Why `concat` on the empty array rather than a partially built array?

Reply:
I was trying to avoid concats getting in the way of seeing the values being concatenated.

Actually I think I can do: `['/q', '/k', vsDevCmdPath, ...vsDevCmdArgs, '&&', 'set']`

Reply:
That seems much better.